components/project/sections/TransactionsSection.vue

```diff
@@ -174,53 +164,22 @@ const handleTagsResponse = (response: TagSearchResponseDTO | null) => {
   tagOptions.value = tagQuery.value.pageNumber === 1 ? newTagOptions : [...tagOptions.value, ...newTagOptions];
 };
 
-const handleTransactionQuery = (query: TransactionSearchRequestDTO) => {
-  const serializedTagNames = searchValues.value.tagNames.join(',');
-  router.push({ query: { ...query, ...(serializedTagNames ? { tagNames: serializedTagNames } : {}) } });
-};
-
-const handleTransactionsResponse = (response: TransactionSearchResponseDTO | null) => {
-  transactionPageCount.value = response?.pageCount || 1;
-  if (transactionQuery.value.pageNumber > transactionPageCount.value) {
-    transactionQuery.value.pageNumber = pagination.page = transactionPageCount.value;
-  }
+const handleTransactionsPaginationQuery = (query: PaginationRequestDTO) => {
+  transactionsQuery.value.pageNumber = query.pageNumber;
+  transactionsQuery.value.pageSize = query.pageSize;
 };
 
 const handleTransactionsSearch = () => {
-  transactionQuery.value = { ...transactionQuery.value, ...searchValues.value, pageNumber: 1 };
-  pagination.page = 1;
+  transactionsQuery.value = { ...transactionsQuery.value, ...searchValues.value, pageNumber: 1 };
+  resetPageNumber();
 };
 
-const handleTransactionPageChange = (pageNumber: number) => {
-  transactionQuery.value.pageNumber = pagination.page = pageNumber;
-};
-
-const handleTransactionPageSizeChange = (pageSize: number) => {
-  transactionQuery.value.pageSize = pagination.pageSize = pageSize;
-};
-
-const pagination = reactive({
-  page: transactionQuery.value.pageNumber,
-  pageSize: transactionQuery.value.pageSize,
-  showSizePicker: true,
-  pageSizes: transactionPageSizes,
-  pageCount: transactionPageCount,
-  onChange: handleTransactionPageChange,
-  onUpdatePageSize: handleTransactionPageSizeChange,
-  simple: false,
-});
-
-useResizeObserver(table, (entries) => {
-  pagination.simple = entries[0].contentRect.width <= 530;
+onMounted(() => {
+  updatePageCount(transactionsData.value, false);
+  pushQueryToUrl(transactionsQuery.value);
 });
-
-const handleMounted = () => {
-  transactionPageCount.value = transactionsResponse.value?.pageCount || 1;
-  handleTransactionQuery(transactionQuery.value);
-};
-
 watch(tagsResponse, handleTagsResponse);
-watch(transactionsResponse, handleTransactionsResponse);
-watch(transactionQuery, handleTransactionQuery, { deep: true });
-onMounted(handleMounted);
+watch(transactionsData, (data) => updatePageCount(data));
```

Comment:
No need for arrow function

Reply:
Has to be arrow function, because the implementation of the expected function is a little different. The watcher expects a function (newValue, oldValue), but the updatePageCount function has params (newValue, adjustPageNumber: boolean)